streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -602,7 +603,7 @@
                     REPLICATION_FACTOR_DOC)
             .define(STATE_DIR_CONFIG,
                     Type.STRING,
-                    "/tmp/kafka-streams",
+                    Utils.getTempDir().resolve("kafka-streams").toAbsolutePath().toString(),
```

Comment:
Thanks for this @dongjinleekr !

It's kind of a bummer that we actually create this file while defining the configurations, since it results in leaking temp directories, even if users set this config to a different value.

Do you think we can do this in a lazy fashion?

Reply:
Good point John. Maybe we should just use `System.getProperty("java.io.tmpdir")` here?

We have already code in place in `StateDirectory` constructor to create the `./kafka-streams` sub-directory. Might be good enough without any additional changes?

Reply:
@vvcephei @mjsax Totally agree. Since I am now finalizing KAFKA-8403 now, I will fix this with a integration test after that. 🙏